word_guess.rb

```diff
@@ -0,0 +1,163 @@
+#word guess game
+
+#
+require "colorize"
+
+class Word_guess
+	attr_accessor :wrong_count, :word_list, :secret_word, :game_word, :game_word_array
+
+	def initialize
+		@wrong_count = -1
+		@word_list = nil
+		@secret_word = nil
+		@game_word = nil
+		@game_word_array = nil
+		@guessed_letters = []
+	end
+
+	def word_gen
+		@word_list = {
+		"celebrity" => ["sting", "beyonce", "madonna", "bono", "fabio", "nas", "prince", "adele","beck", "tupac"],
+		"difficult" => ["ghost","pacman","joker", "bricks", "grandmother", "ninja", "fuzziest", "jazzy", "dwarves", "buffalo"],
+		"animals" => ["monkey", "panda", "cat", "anteater", "girrafe", "elephant", "hotdog", "tiger"]
+		}
+
+		puts "Please pick a theme: celebrity, animals or difficult"
+		theme_input = gets.chomp.downcase
+		until theme_input == "celebrity" || theme_input == "difficult" || theme_input == "animals"
+			puts "Wrong. Please pick one of the following: celebrity, animals or difficult"
+			theme_input = gets.chomp.downcase
+		end
+
+		@secret_word = @word_list[theme_input].sample
+		@game_word = ("_"*@secret_word.length)
+
+		@secret_word_array = @secret_word.split(//)
+		@game_word_array = @game_word.split(//)
+	end
+
+	#asks user for a letter to guess in the word
+	def user_input
+		puts "Please pick a letter to guess the word."
+		@letter = gets.chomp.downcase
+		@letters_array = [*?a..?z] #creates an array of the alphabet from a to z lowercase
+		user_input_check
+	end
+
+	def user_input_check
+		until @letters_array.include?(@letter)
+			puts "Invalid Input".colorize(:red)
+			user_input
+		end
+
+		if @guessed_letters.include?(@letter) || @game_word_array.include?(@letter)
+			puts %!You've already used "#{@letter}"!.colorize(:red)
```

Comment:
Dang yinz are good at using `String#%` to do string formatting. I'm guessing this line came from wanting to use both interpolation and literal double quotes inside the string. Using `%` is a really creative solution. Nicely done! Another way to do it would be:

```ruby
puts "You've already used \"#{@letter}\"".colorize(:red)
```